src/ServiceStack.Azure/Messaging/ServiceBusMqMessageFactory.cs

```diff
@@ -124,28 +128,51 @@ protected internal void StopQueues()
             sbClients.Clear();
         }
 
-        protected internal QueueClient GetOrCreateClient(string queueName)
+        protected internal QueueClient GetOrCreateClient(string queueName, ReceiveMode receiveMode = ReceiveMode.PeekLock)
         {
             queueName = queueName.SafeQueueName();
-            
+
             if (sbClients.ContainsKey(queueName))
                 return sbClients[queueName];
 
-#if !NETSTANDARD2_0
-            // Create queue on ServiceBus namespace if it doesn't exist
-            var qd = new QueueDescription(queueName);
-            if (!namespaceManager.QueueExists(queueName))
-                namespaceManager.CreateQueue(qd);
-#endif
+            var qd = RegisterQueueByName(queueName);
 
 #if NETSTANDARD2_0
-            var sbClient = new QueueClient(address, queueName);
+            var sbClient = new QueueClient(address, qd.Path, receiveMode);
 #else
-            var sbClient = QueueClient.CreateFromConnectionString(address, qd.Path);
+            var sbClient = QueueClient.CreateFromConnectionString(address, qd.Path, receiveMode);
 #endif
-
+            sbClient.PrefetchCount = PrefetchCount;
             sbClient = sbClients.GetOrAdd(queueName, sbClient);
             return sbClient;
         }
+
+        protected internal QueueDescription RegisterQueueByName(string queueName)
+        {
+            var mqDesc = new QueueDescription(queueName);
+#if !NETSTANDARD2_0
+                    if (!namespaceManager.QueueExists(queueName))
+                        namespaceManager.CreateQueue(mqDesc);
+#endif
+#if NETSTANDARD2_0
+            try
+            {
+                managementClient.QueueExistsAsync(queueName)
```

Comment:
@mythz wouldn't mind some guidance on how you'd like to see this part done. All management apis are async so went this route for now

Reply:
The async method would normally be propagated back to the public API which would return a Task Response. But if you want to keep the different implementations the same as the sync API you would block and call `.Wait()` and `.Result` however this opens the chance of a deadlock.

To best mitigate the changes, ideally registering queue's would only be done at registration for all the registered queue's when the MQ is started.